### PR TITLE
MM-15494 Adding warning log when someone has configured a Mattermost push proxy with HTTP endpoint

### PR DESCRIPTION
**Pull request by stylianosrigas**
Adding warning log when someone has configured a Mattermost push proxy with HTTP endpoint

JIRA ticket -> https://mattermost.atlassian.net/browse/MM-15494

### CODE REVIEW

File: app/server.go [hunk 1]
```diff
@@ -216,6 +216,8 @@ func NewServer(options ...Option) (*Server, error) {
 	mlog.Info(fmt.Sprintf("Current working directory is %v", pwd))
 	mlog.Info("Loaded config", mlog.String("source", s.configStore.String()))
 
+	s.PushHostChecker()
```
Comment (jespino) [inline app/server.go]: I don't love the name, and I think it should be private to the server (nobody need to access to this from the outside).

Probably something like `checkPushNotificationServerUrl`, is more explicit about what the function does.

**Comment by DSchalla** [comment]
We maybe need to create a follow up ticket to update the warning text when we fully migrated to config in the database, but otherwise looks good :+1:

**Comment by jespino** [comment]
@DSchalla Good point, I think we already have full support of database configuration, so a change in the text to something more generic would be great. Something like "Your push notification server is configured with HTTP. For improved security, update to HTTPS in your configuration."

**Comment by stylianosrigas** [comment]
Thanks for the feedback @jespino @DSchalla. Warning message updated ;)

**Pull request merged**